modules-provider/src/main/java/org/jahia/modules/external/modules/ModulesDataSource.java

```diff
@@ -222,6 +227,17 @@ public void process(FileMonitorResult result) {
                     logger.debug(result.getInfo());
                 }
                 boolean nodeTypeLabelsFlushed = false;
+
+                Bundle bundle = BundleUtils.getBundleBySymbolicName("external-provider-modules", null);
+                ServiceReference<?>[] allServiceReferences;
+                try {
+                    allServiceReferences = bundle.getBundleContext().getAllServiceReferences(ModulesSourceMonitor.class.getName(), null);
+                } catch (InvalidSyntaxException e) {
+                    if (logger.isDebugEnabled()) {
+                        logger.error(e.getMessage(), e);
+                    }
+                    allServiceReferences = null;
+                }
```

Comment:
Can't you add the references directly through an osgi:list in the ModulesDataSourcePrototype bean ?

Reply:
Yes it is working will submit/update once tested by other team mebers